pkg/hookbot/hookbot.go

```diff
@@ -115,12 +115,19 @@ func (h *Hookbot) Shutdown() {
 	h.wg.Wait()
 }
 
-// Returns "true" if fullTopic ends with `?recursive` and returns topic name
-// without `?recursive` suffix.
+// Returns "true" if fullTopic ends with a "/".
+// Also supports topics ending with ?recursive to support legacy infrastructure.
 func recursive(fullTopic string) (topic string, isRecursive bool) {
+	// Legacy: topics ending in ?recursive are recursive.
```

Comment:
As this project is not versioned yet and we are likely the only users, I vote for removing backwards-compatibility if we don't have anything using and/or can change it at the same time.